apps/evm/src/containers/AssetAccessor/index.tsx

```diff
@@ -1,19 +1,24 @@
 /** @jsxImportSource @emotion/react */
-import { useGetPool } from 'clients/api';
+import { type GetBalanceOfOutput, useGetBalanceOf, useGetPool } from 'clients/api';
 import { Spinner } from 'components';
 import { ConnectWallet } from 'containers/ConnectWallet';
 import { useAccountAddress } from 'libs/wallet';
 import type { Asset, Pool, TokenAction, VToken } from 'types';
 import { areTokensEqual } from 'utilities';
 
+import { useIsFeatureEnabled } from 'hooks/useIsFeatureEnabled';
 import DisabledActionNotice from './DisabledActionNotice';
 
 export interface AssetAccessorProps {
   vToken: VToken;
   poolComptrollerAddress: string;
   connectWalletMessage: string;
   action: TokenAction;
-  children: (props: { asset: Asset; pool: Pool }) => React.ReactNode;
+  children: (props: {
+    asset: Asset;
+    pool: Pool;
+    userWalletNativeTokenBalanceData?: GetBalanceOfOutput;
```

Comment:
I think this extends the container too much, as it is only meant to fetch a specific asset (the native balance doesn't really fit in that context).

I'd suggest creating a separate container to fetch the native balance then wrap any component that needs this data with it.